aiohttp/client_reqrep.py

```diff
@@ -573,6 +587,9 @@ def __init__(self, method, url, *,
         self._auto_decompress = auto_decompress
         self._cache = {}  # reqired for @reify method decorator
 
+        # avoid circular reference so that __del__ works
+        self.on_chunk_received = Signal(owner=None)
```

Comment:
Not sure if we need a signal here.
It looks like signal concept overusage: you are creating `on_chunk_received` signal with the only usage by aiohttp internals itself. It is not very effective, signal subscription is literally a waste of time.
I suggest passing `traces` directly to response object.

Reply:
Here the approach with passing `traces` will actually work quite well. I've introduced the signal to make it symetrical to the situation in request. I will update this to pass `traces` and remove the signal